### PR TITLE
[Windows] Add pipx and install yamllint through it

### PR DESCRIPTION
**Pull request by nikita-bykov**
# Description
Added pipx tool on Windows and installed yamllint through it

#### Related issue:
#1657 
## Check list
- [x] Related issue / work item is attached
- [x] Tests are written (if applicable)
- [x] Documentation is updated (if applicable)
- [x] Changes are tested and related VM images are successfully generated
## Builds
[Windows2016](https://github.visualstudio.com/virtual-environments/_build/results?buildId=87668&view=results), [Windows2019](https://github.visualstudio.com/virtual-environments/_build/results?buildId=87422&view=results)

### CODE REVIEW

File: images/win/scripts/Installers/Install-Pipx-Packages.ps1 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+foreach($tool in $pipxToolset) {
+    if ($tool.python) {
+        $pythonPath =  "${env:AGENT_TOOLSDIRECTORY}\Python\${tool.python}*\${env:AGENT_OSARCHITECTURE}\python-${tool.python}*"
+        Write-Host "Install ${tool.package} into python ${tool.python}"
+        pipx install $tool.package --python $python_path
+    } else {
+        Write-Host "Install ${tool.package} into default python"
+        pipx install $tool.package
+    }
+    Invoke-PesterTests -TestFile "PipxPackages" -TestName $tool.package
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Install-Pipx-Packages.ps1:11]: Why do we need to run it inside loop? `Invoke-PesterTests`

File: images/win/scripts/Tests/PipxPackages.Test.ps1 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Describe "yamllint" {
```
Comment (maxim-lobanov) [inline images/win/scripts/Tests/PipxPackages.Test.ps1:1]: Can we iterate through toolset and generate tests in runtime?

File: images/win/scripts/Installers/Install-Pipx-Packages.ps1 [hunk 3]
```diff
@@ -0,0 +1,12 @@
+$pipxToolset = (Get-ToolsetContent).pipx
+foreach($tool in $pipxToolset) {
+    if ($tool.python) {
+        $pythonPath =  "${env:AGENT_TOOLSDIRECTORY}\Python\${tool.python}*\${env:AGENT_OSARCHITECTURE}\python-${tool.python}*"
```
Comment (maxim-lobanov) [inline images/win/scripts/Installers/Install-Pipx-Packages.ps1:4]: 1) `$pythonPath` vs `$python_path`
2) `"${env:AGENT_TOOLSDIRECTORY}\Python\${tool.python}.*\x64\python-${tool.python}.*"`
3) You need `Get-Item` to get exact path

**Review by vsafonkin — APPROVED** [review]

**Review by vsafonkin — APPROVED** [review]

File: images/win/toolsets/toolset-2019.json [hunk 4]
```diff
@@ -290,5 +290,11 @@
             "mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2019",
             "microsoft/aspnetcore-build:1.0-2.0"
         ]
-    }
+    },
+    "pipx": [
+        {
+            "package": "yamllint",
+            "cmd": "yamllint --version"
```
Comment (maxim-lobanov) [inline images/win/toolsets/toolset-2019.json:297]: Do we really need cmd field here? can we just use `$package --version`?

File: images/win/windows2019.json [hunk 5]
```diff
@@ -326,6 +325,16 @@
                 "{{ template_dir }}/scripts/Installers/Install-CodeQLBundle.ps1"
             ]
         },
+        {
+            "type": "powershell",
+            "environment_vars": [
+                "TOOLSET_JSON_PATH={{user `toolset_json_path`}}"
+            ],
+            "scripts": [
```
Comment (maxim-lobanov) [inline images/win/windows2019.json:333]: Please avoid creating of new section. Use the existing one. Please put these scripts where `Install-YAMLLint.ps1` was run previously

Comment (nikita-bykov) [reply images/win/windows2019.json]: Moved it

**Review by MaksimZhukov — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Pull request merged**